### PR TITLE
prefers-color-scheme example - attempt 2

### PR DESCRIPTION
**Pull request by NeatNit**
### Description

New attempt at #23911. Make an example that more closely resemble real-world usage of the feature. Add detailed description of what the example demonstrates, how it works, and why. Follow the [guidelines](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples#traditional_live_samples) for live samples: add sub-sections for HTML, CSS and Result, and add prose explaining the result.

### Motivation

The [current example](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme#detecting_a_dark_theme) is confusing and lacks explanations. As discussed in #23911, at minimum, it's required to add explanations as well as rename the themes from `day` and `night` to something not associated with lightness and darkness.

In trying to do that, I realized that it can be modified and improved more subtly than my previous attempt, while having better real-world applications. [To quote the guidelines](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples#general_guidelines):

> Go beyond just demonstrating how each piece of the API or technology works. Consider possible real-world use cases you might try to demonstrate.

As such, the adapted example demonstrates exactly how a real website might use `prefers-color-scheme` to style itself. Please look at it to see how.

### Additional notes

I welcome any feedback and suggestions, especially on the exact colors of Theme A and Theme B. I've checked `allow edits by maintainers`, so if you are a maintainer you can edit the colors directly. For any other change (e.g. in phrasing), I would prefer if you suggest changes here through discussion. 🙂

### CODE REVIEW

**Comment by NeatNit** [comment]
@estelle hi, I don't mean to nag but could you take a look? Or at least allow the bot to generate a preview?

**Comment by estelle** [comment]
> @estelle hi, I don't mean to nag but could you take a look? Or at least allow the bot to generate a preview?

The preview is available at https://pr24495.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/@media/prefers-color-scheme. I was out since Friday. Apologies that no one else took a look at this while I was away. Will look at it today.

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 1]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
+  height: 2em;
+  vertical-align: middle;
 }
+```
 
+Theme A (brown) uses a light color scheme by default, but will switch to a dark scheme based on the media query:
+```css
+.theme-a {
+    background: #dca;
+    color: #731;
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:71]:
```suggestion
    color: #731;
    outline: 5px solid #000;
```

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 2]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
-  .day.dark-scheme {
-    background: #333;
-    color: white;
-  }
-  .night.dark-scheme {
-    background: black;
-    color: #ddd;
+  .theme-a.adaptive {
+      background: #753;
+      color: #dcb;
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:76]:
```suggestion
      color: #dcb;
      outline: 5px double currentcolor;
```

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 3]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
   }
 }
+```
 
+Theme B (blue) uses a dark color scheme by default, but will switch to a light scheme based on the media query:
+```css
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:82]:
```suggestion
Theme B (blue) uses a dark color scheme by default, but will switch to a light scheme based on the media query:

```css
```

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 4]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
+.theme-b {
+    background: #447;
+    color: #bbd;
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:85]:
```suggestion
    color: #bbd;
    outline: 5px solid #000;
```

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 5]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
-  .day.light-scheme {
-    background: white;
-    color: #555;
-  }
-  .night.light-scheme {
-    background: #eee;
-    color: black;
+  .theme-b.adaptive {
+      background: #bcd;
+      color: #334;
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:90]:
```suggestion
      color: #334;
      outline: 5px dotted currentcolor;
```

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 6]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
-  width: 7em;
-  height: 2em;
-  vertical-align: middle;
-}
 ```
 
-{{EmbedLiveSample("Detecting_a_dark_theme")}}
+#### Result
+
+The left boxes shows Theme A and Theme B as they would appear without the `prefers-color-scheme` media query. The right boxes show the same themes, but one of them will be changed to a darker or lighter variant based on the user's active color scheme.
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:97]:
```suggestion
The left boxes shows Theme A and Theme B as they would appear without the `prefers-color-scheme` media query. The right boxes show the same themes, but one of them will be changed to a darker or lighter variant based on the user's active color scheme. The outline of one box will be doubled or dotted based on your browser or operating systems settings.
```

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 7]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
+div.box {
+  display: inline-block;
+  padding: 1em;
+  width: 7em;
+  height: 2em;
+  vertical-align: middle;
 }
+```
 
+Theme A (brown) uses a light color scheme by default, but will switch to a dark scheme based on the media query:
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:67]:
```suggestion
Theme A (brown) uses a light color scheme by default, but will switch to a dark scheme based on the media query:

```

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 8]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
+This example shows both options: Theme A uses light colors, but can be overridden to dark colors. Theme B uses dark colors, but can be overridden to light colors. In the end, if the browser supports `prefers-color-scheme`, both themes will be light or dark.
+
+#### HTML
 
 ```html
-<div class="day">Day (initial)</div>
-<div class="day light-scheme">Day (changes in light scheme)</div>
-<div class="day dark-scheme">Day (changes in dark scheme)</div>
+<div class="box theme-a">Theme A (initial)</div>
+<div class="box theme-a adaptive">Theme A (adaptive)</div>
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:48]:
```suggestion
<div class="box theme-a adaptive">Theme A (changed if dark preferred)</div>
```

File: files/en-us/web/css/@media/prefers-color-scheme/index.md [hunk 9]
```diff
@@ -35,67 +35,68 @@ To learn more about SVGs, see the [SVG documentation](/en-US/docs/Web/SVG) and f
...
 <br />
 
-<div class="night">Night (initial)</div>
-<div class="night light-scheme">Night (changes in light scheme)</div>
-<div class="night dark-scheme">Night (changes in dark scheme)</div>
+<div class="box theme-b">Theme B (initial)</div>
+<div class="box theme-b adaptive">Theme B (adaptive)</div>
```
Comment (estelle) [inline files/en-us/web/css/@media/prefers-color-scheme/index.md:52]:
```suggestion
<div class="box theme-b adaptive">Theme B (adaptive)</div>
```
```suggestion
<div class="box theme-b adaptive">Theme B (changed if light preferred)</div>
```

**Review by estelle — COMMENTED** [review]
I  suggested adding outline changes for the user's browser preference, and changed the word "adaptive" to "changed if X preferred" b/c i don't think all users, especially non-native English readers will full understand  that. 

(I also committed to white space issues that prettier requires if you're wondering what my commit are.)

Look good!

**Comment by NeatNit** [comment]
> I suggested adding outline changes for the user's browser preference, and changed the word "adaptive" to "changed if X preferred" b/c i don't think all users, especially non-native English readers will full understand that.
> 
> (I also committed to white space issues that prettier requires if you're wondering what my commit are.)
> 
> Look good!

Thank you! I am just going to bed but will give this a whirl tomorrow.

You're right that "changed if X preferred" is clearer.  I think the boxes aren't big enough to fit that text - will have to check in both Firefox and Chromium. But it should be easy to make the boxes bigger or the text smaller.

I like the outline idea, but I would make it dotted in both cases and phrase the text: "The theme that was changed has a dotted outline." I think that would be clearer, do you agree?

**Comment by estelle** [comment]
> I like the outline idea, but I would make it dotted in both cases and phrase the text: "The theme that was changed has a dotted outline." I think that would be clearer, do you agree?

People learn in different ways. I think having clearly different outlines is helpful to those who learn best via visual cues.

**Comment by NeatNit** [comment]
After my tweaks, this is what the output looks like (via jsfiddle):

Light theme:
Firefox:
Firefox
Vivaldi (chromium):
Vivaldi

Dark theme:
Firefox:
Firefox
Vivaldi:
Vivaldi

In my opinion, the double-outline in dark Theme A looks too pale and unclear. On the other hand, if I change both adaptive rules to `outline: 5px dotted #000;`, we get:

Light theme (Firefox):
Firefox
Dark theme (Firefox):
Firefox
(Vivaldi doesn't differ meaningfully, other than the font differences already visible above)

The dotted outline is easier to notice. The description, then, can read:
> The outline of one box will be dotted, indicating that it was changed based on your browser or operating systems settings.

Would you accept that?

Thank you either way :)

**Comment by NeatNit** [comment]
> People learn in different ways. I think having clearly different outlines is helpful to those who learn best via visual cues.

Just to respond more directly - I absolutely agree. I just think it's better/clearer visually to say "a dotted outline means the theme changed", as opposed to "if a theme changed from light to dark then it will have a doubled outline but if it changed from dark to light then it will have a dotted outline". I liked your idea and implementation, the only difference I'm suggesting is to use the same rule you made for Theme B and apply it to Theme A as well.

**Comment by estelle** [comment]
You're right. The double border is barely noticeable. Dotted is super noticable.

**Comment by NeatNit** [comment]
> You're right. The double border is barely noticeable. The dotted is super noticeable. Are you OK with changing the `double` to `dashed` (so one is `dotted` and the other `dashed`) just to better enable people to see which code block is being used?

I now see what you originally meant - sorry, it went a bit over my head before, I was getting stuck on the double border being barely noticeable.

Yes, I think a dashed outline for the dark Theme A would be the best solution. I'll try to make the change within the next hour and share a screenshot of the result.

**Comment by NeatNit** [comment]
Note:
1. The first CSS block defining div.box is *hidden* and would not appear on MDN. This is already visible (or rather, invisible) in the preview: https://pr24495.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/@media/prefers-color-scheme - it has been that way from the initial PR, I just wanted to highlight it so there's no miscommunication :)
  The reason I decided to hide it is based on the linked guidelines - it's very boring and just defines a box shape. The class name is enough to convey what it does, and it has no bearing on the article's subject - the color scheme.
2. All outlines are black - done for readability with dark Theme A (where currentColor is too light) and then for consistency matched on light Theme B.

Looks good to me! I hope you agree, and I'm all ears if not.

**Review by estelle — APPROVED** [review]
Thanks for all the work on this! Looks good.

It's fine that the extra CSS is not visible, as long as the important to the topic of the page lines are visible. Thank you!!

**Comment by NeatNit** [comment]
Thank you for all the cooperation! ❤️

**Pull request merged**